File: oauth2-server/src/main/java/a/gleb/keycloaktojwt/db/entity/OAuth2AuthorizationInner.java
package a.gleb.keycloaktojwt.db.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.*;

import java.time.Instant;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "oauth2_authorization")
public class OAuth2AuthorizationInner {

    @Id
    @Column(name = "id", nullable = false)
    private String id;

    @Column(name = "principal_name")
    private String principalName;

    @Column(name = "access_token_issued_at")
    private Instant accessTokenIssuedAt;
}

File: oauth2-server/src/main/kotlin/a/gleb/oauth2server/db/entity/OAuth2AuthorizationInner.kt
package a.gleb.oauth2server.db.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType.AUTO
import jakarta.persistence.Id
import java.time.Instant

@Entity(name = "oauth2_authorization")
data class OAuth2AuthorizationInner(

    @Id
    @GeneratedValue(strategy = AUTO)
    var id: String?,

    @Column(name = "principal_name")
    var principalName: String,

    @Column(name = "access_token_issued_at")
    var accessTokenIssuedAt: Instant
)
